chapters/functions.tex

```diff
@@ -1745,9 +1745,19 @@ \subsubsection{Simple Types}\label{simple-types}
 
 An exception is made when the argument is of the form \lstinline!size($\ldots$, $\ldots$)!. In this case the corresponding C type is \lstinline!size_t!.
 
-Strings are \textsc{nul}-terminated (i.e., terminated by \lstinline[language=C]!'\0'!) to
-facilitate calling of C functions. When returning a non-literal string,
-see \cref{utility-functions-for-allocating-strings} for details on memory allocation.
+Strings are \textsc{nul}-terminated (i.e., terminated by \lstinline[language=C]!'\0'!) to facilitate calling of C functions.
+The valid return values for an external function returning a \lstinline!String! are:
+\begin{itemize}
+\item A string given as \lstinline!String! input to the external function.
+\item A pointer to a C string literal.
+\item A pointer returned by one the string allocation functions in \cref{utility-functions-for-allocating-strings}.
+\end{itemize}
+
+\begin{nonnormative}
+The reason why it is not allowed to return a string allocated with, for instance, \lstinline[language=C]!malloc! is that there is no transfer of ownership when a string is returned from the external function.
+The external code would remain the owner of such a string, and would be responsible for eventually releasing the memory at some point.
+Consequently, the Modelica simulation environment would not be able to assume that all only its own string deallocation routines could invalidate any of the strings returned by external functions.
```

Comment:
Some words are missing here.

Reply:
I think the problem is that there's one _all_ too much:
```suggestion
Consequently, the Modelica simulation environment would not be able to assume that only its own string deallocation routines could invalidate any of the strings returned by external functions.
```